compiler/rustc_trait_selection/src/traits/query/type_op/implied_outlives_bounds.rs

```diff
@@ -202,7 +202,17 @@ pub fn compute_implied_outlives_bounds_compat_inner<'tcx>(
                 }
             }
 
-            let pred = match obligation.predicate.kind().no_bound_vars() {
+            let pred = if ocx.infcx.next_trait_solver() {
```

Comment:
r=me, however, do you think it would be cleaner to always use `wf::obligations_unnormalized` and just use `deeply_normalize` in both solvers?

That way we shouldn't need the above `ocx.register_obligation(obligation.clone());`